Overlap issue

hepcat72 opened the pull request:
**THIS WAS MERGED INTO `signed_arrowhead`, so this should be closed after the description has been read.**

Resolves issue #42

Note that the step sizes set in the bezier edges may not be necessary or match the unbundled-bezier control-point-distances.  I was just happy to get 2 tiers of bowing working.

Unbundled-bezier is used in 2 classes (one for each tier of bowing) is used when a gap junction is present to prevent overlap.

Note that each tier is meant to contain 1 connection type where feasible.  Default in the full case is synapses on the inner tier and funcons in the outer.

Ostensibly, this should have been able to work with a single class with 4 control-point-distances in a string (without the ability to control which types occur in which tiers), but every time I tried that, there was only 1 tier of bowing and overlap would occur.

Please see if you notice a difference in speed or if bowing looks different with & without gap junctions present.  Also not that without gap junctions, everything is controlled by bezier, as it was originally.

Also - I did not endeavor to silence impossible-to-draw warnings.  At various points during my working, they would go away and then come back.  I believe they have to do with lines overlapping in the initial attempt to draw and that it falls back to a "parent method" (my hypothesis).

hepcat72 committed:
Had to add some complexity to the bowing logic since there are 3 types of lines now and there are multiple combinations of possible neighboring line types.  The previous logic only worked with 2.

Files checked in: src/client/js/view/network-style.js src/client/js/view/network.js

hepcat72 committed:
I finally got the console output to be predictable in the case where there are 2 synapses and 1 functional connection and a gap junction.  The connections change on hover and drag, but at least I know how to loop on all parallel edges.

Files checked in: src/client/js/view/network-style.js src/client/js/view/network.js
[skip ci]

hepcat72 committed:
Looks like using unbundled-bezier works and I got the distances format worked out.  This commit only does this for cases where there are an odd number of lines WITH a gap junction present.  I should be able to simplify this in the next commit.

Files checked in: src/client/js/view/network-style.js
[skip ci]

hepcat72 committed:
Checking this in after undoing a few changes that I thought would be completely innocuous (just variable name changes), but the bad behavior came back.  This version *appears* to completely work for AUA connections.  Nothing changes on hover.  But there's a bunch of unused code and the variable names do not accurately reflect the data they represent.

Files checked in: src/client/js/view/network-style.js src/client/js/view/network.js
[skip ci]

hepcat72 committed:
This resolves #42.

unbundled-bezier, used in 2 classes (one for each tier of bowing) is used when a gap junction is present to prevent overlap.

Note that each tier is meant to contain 1 connection type where possible.  Default in the full case is synapses on the inner tier and funcons in the outer.

Ostensibly, this should have been able to work with a single class with 4 control-point-distances in a string, but every time I tried that, there was only 1 tier of bowing and overlap would occur.

Files checked in: src/client/js/view/network-style.js src/client/js/view/network.js

hepcat72 committed:
Forgot to remove a vestigial class that was unused.

Files checked in: src/client/js/view/network-style.js

hepcat72 committed:
Lints.

Files checked in: src/client/js/view/network.js

hepcat72 committed:
Compared images with neuron-graph.princeton.edu and realized I had too much bowing for the no-GJ cases.

Files checked in: src/client/js/view/network-style.js

lparsons reviewed (approved):
This looks great to me, and fixed the cases I could find. Excellent work!

hepcat72 committed:
Tweak arrow in legend (#46)

* Tweak arrow in legend.

Files checked in: src/client/scss/legend.scss

* Retain the icon-functional label.

Files checked in: src/client/scss/icons.scss src/client/scss/legend.scss
